sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -271,6 +315,139 @@ def _configure_timeout(request: PipelineRequest, absolute: Optional[int], per_re
         # Only socket timeout provided.
         request.context.options['connection_timeout'] = per_request
 
+def __populate_logger_attributes(logger_attributes={}, request=None, response_headers=None, exception=None):
```

Comment:
we should probably move all of these logging methods into someplace more relevant, or another file altogether - `_retry_utility.py` shouldn't house this logic